### DISCUSSION TITLE
Is there any way to provide multi line parameter in a Step?

### DISCUSSION THREAD

**Discussion by samuelbenison** [getgauge/gauge#2160] [General]
Hi, I would like to pass a multi-line string as a parameter via step? Is that possible?

For example, Something like this:

```
* A step with string "str1" and a multi-line string "
{
  "str2": "something",
  "str3": "something else"
}
"
```

My goal is to pass a dictionary (python) as a parameter, but the items may get a little large, and I don't want to crowd them in a single unformatted line, then it might become a little hard to read. It is okay if I can only be able to pass them as a string, I can convert it into a dictionary later.

Can you please suggest what are my options to pass JSON like data in a step parameter? Thanks

**Comment by zabil** [comment]
Multi-line parameters are not supported at the moment. If it's a dictionary have you tried passing it as a [table parameters](https://docs.gauge.org/writing-specifications.html?os=macos&language=javascript&ide=vscode#table-parameters)?

You can then loop through these values in the step.

**Comment by karimnabli** [comment]
this can be closed ? 
https://github.com/getgauge/gauge/pull/2788
now we support multiline string,

**Reply by chadlwilson** [reply]
This is just a discussion, so there's nothing to close - but yes :-)
